src/components/Image/Image.visual.tsx

```diff
@@ -29,8 +30,11 @@ type ImageWithWrapperProps = ImageProps & {
 class ImageWithWrapper extends React.Component<ImageWithWrapperProps> {
   state = { hasError: false };
 
-  onError(onError) {
-    this.setState({ hasError: true }, () => setTimeout(() => onError(), 500));
+  async onError(onError) {
+    this.setState({ hasError: true }, async () => {
+      await delay(500);
+      onError();
+    });
```

Comment:
Isn't it problematic that `onError` is defined twice?
```suggestion
  async onError(callback) {
    this.setState({ hasError: true }, async () => {
      await delay(500);
      callback();
    });
```

Reply:
Oops! I did the fix in a different PR instead 🤦🏻‍♂️:
https://github.com/wix/wix-ui-tpa/pull/651/files#diff-59cb9bbb8734f702d7df647b858b4173e292160bcc9615bbb390a3eecd5a7aabR32

I just pushed a fix to this branch.